plugins/kotlin/code-insight/intentions-shared/src/org/jetbrains/kotlin/idea/codeInsight/intentions/shared/utils/FillInArgumentsUtils.kt

```diff
@@ -0,0 +1,97 @@
+// Copyright 2000-2023 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
+package org.jetbrains.kotlin.idea.codeInsight.intentions.shared.utils
+
+import com.intellij.psi.util.elementType
+import org.jetbrains.kotlin.idea.base.psi.getLineNumber
+import org.jetbrains.kotlin.idea.references.mainReference
+import org.jetbrains.kotlin.lexer.KtTokens
+import org.jetbrains.kotlin.name.Name
+import org.jetbrains.kotlin.psi.KtCallElement
+import org.jetbrains.kotlin.psi.KtFunction
+import org.jetbrains.kotlin.psi.KtParameter
+import org.jetbrains.kotlin.psi.KtPsiFactory
+import org.jetbrains.kotlin.psi.KtValueArgumentList
+
+internal object FillInArgumentsUtils {
+
+    fun getFunctionArguments(element: KtValueArgumentList) = element.children.map { it.firstChild.text }.toSet()
+
+    fun isApplicableByPsi(element: KtValueArgumentList, isOptional: Boolean): Boolean {
+        val arguments = getFunctionArguments(element)
+        val callElement = element.parent as KtCallElement
+        val resolvedReference = callElement.calleeExpression?.mainReference?.resolve() ?: return false
+        val paramsChildren = if (resolvedReference is KtFunction) {
+            resolvedReference.valueParameters
+        } else {
+            return false
+        }
+
+        val isMissingMandatoryArguments: Boolean = paramsChildren.any {
+            val ktParam = (it as KtParameter)
+            val parameterName = ktParam.name ?: return@any false
+            !ktParam.hasDefaultValue() && parameterName !in arguments
+        }
+        if (!isOptional) return isMissingMandatoryArguments
+        if (isMissingMandatoryArguments) return false
+
+        return paramsChildren.any {
+            val ktParam = (it as KtParameter)
+            val parameterName = ktParam.name ?: return@any false
+            ktParam.hasDefaultValue() && parameterName !in arguments
+        }
+    }
+
+    fun fillInArguments(element: KtValueArgumentList, argumentsList: List<Name>) {
+        argumentsList.ifEmpty { return }
+        val psiFactory = KtPsiFactory(element.project)
+        val isNewLine = isNewLine(element, argumentsList.size)
```

Comment:
You're passing the number of parameters to `numOfNewArguments`. It's a logical mistake

Reply:
fixed